pennylane/pulse/__init__.py

```diff
@@ -13,6 +13,30 @@
 # limitations under the License.
 """
 This module contains classes and functions used in pulse programming.
```

Comment:
Could be useful to expand this by 1-2 more sentences just so users know _what_ this functionality is useful for? @trbromley what do you think?

Reply:
Expanded to a short intro, as per conversation below